SimTransport/PPSProtonTransport/src/HectorTransport.cc

```diff
@@ -0,0 +1,238 @@
+{
+
+    m_beamline45=nullptr;
+    m_beamline56=nullptr;
+    edm::FileInPath b1(m_beam1filename.c_str());
+    edm::FileInPath b2(m_beam2filename.c_str());
+    if(m_verbosity) {
+        edm::LogInfo("HectorTransportSetup") << "===================================================================\n"  
+            << " * * * * * * * * * * * * * * * * * * * * * * * * * * * *           \n"  
+            << " *                                                         *       \n"  
+            << " *                   --<--<--  A fast simulator --<--<--     *     \n"  
+            << " *                 | --<--<--     of particle   --<--<--     *     \n"  
+            << " *  ----HECTOR----<                                          *     \n"  
+            << " *                 | -->-->-- transport through-->-->--      *     \n"   
+            << " *                   -->-->-- generic beamlines -->-->--     *     \n"  
+            << " *                                                           *     \n"   
+            << " * JINST 2:P09005 (2007)                                     *     \n"  
+            << " *      X Rouby, J de Favereau, K Piotrzkowski (CP3)         *     \n"  
+            << " *       http://www.fynu.ucl.ac.be/hector.html               *     \n"  
+            << " *                                                           *     \n"  
+            << " * Center for Cosmology, Particle Physics and Phenomenology  *     \n"  
+            << " *              Universite catholique de Louvain             *     \n"  
+            << " *                 Louvain-la-Neuve, Belgium                 *     \n"  
+            << " *                                                         *       \n"  
+            << " * * * * * * * * * * * * * * * * * * * * * * * * * * * *           \n"   
+            << " HectorTransport configuration: \n" 
+            << " lengthctpps      = " << m_lengthctpps << "\n"
+            << " m_f_ctpps_f      =  " << m_f_ctpps_f << "\n"
+            << " m_b_ctpps_b      =  " << m_b_ctpps_b << "\n"
+            << "===================================================================\n";
+    }  
+
+    // construct beam line for PPS (forward 1 backward 2):                                                                                           
+    if(m_lengthctpps>0. ) {
+        m_beamline45 = std::unique_ptr<H_BeamLine>(new H_BeamLine(-1, m_lengthctpps + 0.1 )); // (direction, length)
+        m_beamline45->fill( b2.fullPath(), 1, "IP5");
+        m_beamline56 = std::unique_ptr<H_BeamLine>(new H_BeamLine( 1, m_lengthctpps + 0.1 )); //
+        m_beamline56->fill( b1.fullPath(), 1, "IP5");
+    }
+    else {
+        if ( m_verbosity ) LogDebug("HectorTransportSetup") << "HectorTransport: WARNING: lengthctpps=  " << m_lengthctpps;
+        return false;
+    }
+    if (m_verbosity) {
+          std::cout  << "====================================================================\n"
```

Comment:
@mundim @civanch cout is not thread safe, could you please replace with LogInfo or LogVerbatim for instance?